lib/membrane/rtp/vad_utils/is_speaking_estimator.ex

```diff
@@ -0,0 +1,118 @@
+defmodule Membrane.RTP.VadUtils.IsSpeakingEstimator do
+  @moduledoc """
+  Module for estimating if the user is speaking based on
+  "Dominant Speaker Identification for Multipoint Videoconferencing
+  by Ilana Volfin and Israel Cohen"
+  """
+
+  # number of levels inside one immediate interval
+  @n1 1
+  # number of immediate intervals inside one medium interval
+  @n2 10
+  # number of medium intervals inside one long interval
+  @n3 5
+
+  @immediate_score_threshold 0
+  @medium_score_threshold 20
+  @long_score_threshold 20
+
+  @min_levels_length @n1 * @n2 * @n3
+
+  # @max_level 127
+  # @min_level 0
+  @medium_threshold 1
+  @long_threshold 3
+
+  @min_activity_score 1.0e-8
+
+  defp binomial_coefficient(n, k) when k < 0 or k > n, do: 0
+  defp binomial_coefficient(n, k) when k == 0 or n == k, do: 1
+
+  defp binomial_coefficient(n, k) do
+    k = min(k, n - k)
+
+    Enum.to_list(1..k)
+    |> Enum.reduce(1, fn i, acc -> div(acc * (n - i + 1), i) end)
+  end
+
+  defp compute_activity_score(vl, n_r, lambda) do
+    p = 0.5
+
+    score =
+      :math.log(binomial_coefficient(n_r, vl)) +
+        vl * :math.log(p) +
+        (n_r - vl) * :math.log(1 - p) -
+        :math.log(lambda) + lambda * vl
+
+    if score > @min_activity_score, do: score, else: @min_activity_score
+  end
+
+  defp indicator_func(x, threshold) do
+    if x >= threshold, do: 1, else: 0
+  end
+
+  defp count_active(data, threshold) do
+    Enum.reduce(data, 0, fn x, acc -> acc + indicator_func(x, threshold) end)
+  end
+
+  defp compute_interval(littles, sub_list_length, threshold) do
+    littles
+    |> Enum.chunk_every(sub_list_length)
+    |> Enum.map(&count_active(&1, threshold))
```

Comment:
Why do you map the values you won't be using? I wanted my initial suggestion to be
```suggestion
    littles
    |> Enum.chunk_every(sub_list_length)
    |> Enum.map(&count_active(&1, threshold))
    |> hd()
```
as you always take the head in `estimate_is_speaking` but this is equivalent to
```suggestion
    littles
    |> Enum.take(sub_list_length)
    |> count_active(threshold)
```
R U sure this is what you wanted to do?

Reply:
This function is used for computing immediates, mediums and longs. It's true only the head is used for evaluating the activity score of each of them...

 *but*
When the mediums are computed, all the values of immediates are used and when longs are computed all values of mediums are needed. Moreover, `longs` has at most 1 element. 
Therefore I think those changes do not apply here. We can think about optimising this process in future, but as for now I don't seethe merit in this either.